specs/_features/epbs/fork.md

```diff
@@ -0,0 +1,133 @@
+    """
+    Return the fork version at the given ``epoch``.
+    """
+    if epoch >= EPBS_FORK_EPOCH:
+        return EPBS_FORK_VERSION
+    if epoch >= DENEB_FORK_EPOCH:
+        return DENEB_FORK_VERSION
+    if epoch >= CAPELLA_FORK_EPOCH:
+        return CAPELLA_FORK_VERSION
+    if epoch >= BELLATRIX_FORK_EPOCH:
+        return BELLATRIX_FORK_VERSION
+    if epoch >= ALTAIR_FORK_EPOCH:
+        return ALTAIR_FORK_VERSION
+    return GENESIS_FORK_VERSION
+```
+
+## Fork to EPBS
+
+### Fork trigger
+
+TBD. This fork is defined for testing purposes, the EIP may be combined with other 
+consensus-layer upgrade.
+For now, we assume the condition will be triggered at epoch `EPBS_FORK_EPOCH`.
+
+### Upgrading the state
+
+If `state.slot % SLOTS_PER_EPOCH == 0` and `compute_epoch_at_slot(state.slot) == EPBS_FORK_EPOCH`,
+an irregular state change is made to upgrade to ePBS.
+
+```python
+def upgrade_to_epbs(pre: deneb.BeaconState) -> BeaconState:
+    epoch = deneb.get_current_epoch(pre)
+
+    post = BeaconState(
+        # Versioning
+        genesis_time=pre.genesis_time,
+        genesis_validators_root=pre.genesis_validators_root,
+        slot=pre.slot,
+        fork=Fork(
+            previous_version=pre.fork.current_version,
+            current_version=EPBS_FORK_EPOCH,  # [Modified in ePBS]
+            epoch=epoch,
+        ),
+        # History
+        latest_block_header=pre.latest_block_header,
+        block_roots=pre.block_roots,
+        state_roots=pre.state_roots,
+        historical_roots=pre.historical_roots,
+        # Eth1
+        eth1_data=pre.eth1_data,
+        eth1_data_votes=pre.eth1_data_votes,
+        eth1_deposit_index=pre.eth1_deposit_index,
+        # Registry
+        validators=pre.validators,
+        balances=pre.balances,
+        # Randomness
+        randao_mixes=pre.randao_mixes,
+        # Slashings
+        slashings=pre.slashings,
+        # Participation
+        previous_epoch_participation=pre.previous_epoch_participation,
+        current_epoch_participation=pre.current_epoch_participation,
+        # Finality
+        justification_bits=pre.justification_bits,
+        previous_justified_checkpoint=pre.previous_justified_checkpoint,
+        current_justified_checkpoint=pre.current_justified_checkpoint,
+        finalized_checkpoint=pre.finalized_checkpoint,
+        # Inactivity
+        inactivity_scores=pre.inactivity_scores,
+        # Sync
+        current_sync_committee=pre.current_sync_committee,
+        next_sync_committee=pre.next_sync_committee,
+        # Execution-layer
+        # Withdrawals
+        next_withdrawal_index=pre.next_withdrawal_index,
+        next_withdrawal_validator_index=pre.next_withdrawal_validator_index,
+        # Deep history valid from Capella onwards
+        historical_summaries=pre.historical_summaries,
+        # ePBS
+        previous_inclusion_list_proposer=0, # [New in ePBS]
+        previous_inclusion_list_slot=0, # [New in ePBS]
```

Comment:
You need to set these to be the latest proposer/slot so that you can import the first payload, and the payload signature verification has to have an exception to not verify a signature from an IL before the fork.

Reply:
I pushed it. It's now 
```
        previous_inclusion_list_proposer=get_beacon_proposer_index(pre), # [New in ePBS]
        previous_inclusion_list_slot=pre.slot, # [New in ePBS]
```

I wont do the beacon chain changes in this PR but let's make a note of that